### DISCUSSION TITLE
Environment collision (?) using scikit-build with conda-build on Windows

### DISCUSSION THREAD

**Discussion by yosefm** [scikit-build/scikit-build#903] [Q&A]
I'm working on transitioning the [SfePy package ](https://sfepy.org/doc-devel/index.html) to scikit-build ([PR here](https://github.com/sfepy/sfepy/pull/960)). So far I managed to get it building in a shell when running setup.py install, even under a pip isolated build environment. However, building a conda package using conda-build (actually mamba-build, but I don't think that matters) [fails on Windows](https://github.com/conda-forge/sfepy-feedstock/runs/12101587051).

I've narrowed it down to the following: When conda-build runs, it sets up a Visual Studio build environment using vcvarsall.bat. I can fail a shell build by doing this step myself. I understand that vcvarsall is not needed when ninja build is used, but I don't see how I can turn it off or affect it.

Deeper still: under vcvarsall, compiler outputs go to _skbuild.../Release/output....pyd, while without it the 'Release' part is omitted. This fails linking of extensions using other libraries in the build. If I knew what environmentvariable causes it, maybe I could use it in CMake, but I can't find it.

Has anyone tried to build in such settings, and can give me some directions?

**Comment by yosefm** [comment]
worked around this by ensuring Ninja is used for all platforms and settings. Still don't know exactly what it was.

**Reply by henryiii** [reply]
Sorry, missed this! At PyCon, will try to look early next week, please ping me if I forget!
